### PR TITLE
APPLE-50 Create github workflows for build and release

### PR DESCRIPTION
**Pull request by benpbolton**
Automates creating a `$VERSION-built` tag and release for any tag pushed with valid `-v:refname` by:
- Shelling out to DeployBot for build
- Automating the release process

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+on:
+  push:
+    tags:
+    - 'v*'  # Push events to matching v* (eg. v2.0.8)
```
Comment (kevinfodness) [inline .github/workflows/build.yml]: Is it possible to make it literally match `v*.*.*`? Or can we only use one wildcard?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+jobs:
+  build:
+    name: Trigger Build
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: 'curl to build server'
+        shell: bash
+        run: curl -X POST https://alleyinteractive.deploybot.com/webhook/dd3e327ef03a30b9b6f84a8a6293cc21750da4f81fa35886
```
Comment (kevinfodness) [inline .github/workflows/build.yml:17]: Nice

**Review by kevinfodness — APPROVED** [review]
🍣

File: .github/workflows/build.yml (hunk 1 above)
Comment (benpbolton) [reply .github/workflows/build.yml]: Indeed! Shall I?

**Pull request merged**